File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        // 정답 선언 및 초기화
        int answer = 0;
        // a와 b의 홀짝 여부에 따라 다른 점수를 계산하는 if문
        if(a%2 == 1 && b%2 == 1){
            answer += a*a + b*b;
        } else if(a%2 == 1 || b%2 == 1){
            answer += 2 * (a + b);
        } else {
            answer += Math.abs(a - b);
        }
        
        // 정답 리턴
        return answer;
    }
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):
    answer = 0
    even = [2, 4, 6]
    odd = [1, 3, 5]
    
    if a in odd and b in odd:
        answer = a**2 + b**2
    elif a in even and b in even:
        answer = abs(a - b)
    else:
        answer = 2 * (a + b)
    
    return answer